### PR TITLE
Projection Map

### PR DESCRIPTION
**Pull request by lheagy**
Simple projection map to select indices of a model 

cc @fourndo

### CODE REVIEW

File: SimPEG/Maps.py [hunk 1]
```diff
@@ -709,6 +709,51 @@ def adj(v):
...
+    """
+
+    def __init__(self, nP, index, **kwargs):
+        assert isinstance(index, (np.ndarray, slice, list)), (
+            'index must be a np.ndarray or slice, not {}'.format(type(index)))
+        super(Projection, self).__init__(nP=nP, **kwargs)
+
+        if isinstance(index, slice):
+            index = list(range(*index.indices(self.nP)))
+        self.index = index
```
Comment (rowanc1) [inline SimPEG/Maps.py]: does it matter here that index is either a list or a numpy array?

Comment (rowanc1) [inline SimPEG/Maps.py]: Do we want an assertion that `max(index) < nP` just for sanity?

File: SimPEG/Maps.py [hunk 2]
```diff
@@ -709,6 +709,51 @@ def adj(v):
...
+        self._shape = nI, nP = len(self.index), self.nP
+
+        # sparse projection matrix
+        self.P = sp.csr_matrix(
+            (np.ones(nI), (range(nI), self.index)), shape=(nI, nP)
+        )
+
+    def _transform(self, m):
+        return m[self.index]
```
Comment (rowanc1) [inline SimPEG/Maps.py]: `assert len(m) == self.nP, "Ensure that the model has {} elements.".format(self.nP)`

File: SimPEG/Maps.py [hunk 3]
```diff
@@ -709,6 +709,51 @@ def adj(v):
...
+
+    @property
+    def shape(self):
+        """
+        Shape of the matrix operation (number of indices x nP)
+        """
+        return (len(self.index), self.nP)
```
Comment (rowanc1) [inline SimPEG/Maps.py]: `return self._shape` from above?

File: tests/base/test_maps.py [hunk 4]
```diff
@@ -246,5 +246,17 @@ def test_ParametricSplineMap(self):
...
         self.assertTrue(mParamSpline.testVec())
 
+    def test_Projection(self):
+        nP = 10
+        m = np.arange(nP)
+        self.assertTrue(np.all(Maps.Projection(nP, slice(5))*m == m[:5]))
+        self.assertTrue(np.all(Maps.Projection(nP, slice(5, None))*m == m[5:]))
+        self.assertTrue(np.all(Maps.Projection(nP, np.r_[1, 5, 3, 2, 9, 9])*m ==
+                        np.r_[1, 5, 3, 2, 9, 9]))
+        self.assertRaises(Exception, Maps.Projection(nP, [1, 5, 3, 2, 9, 9]))
```
Comment (rowanc1) [inline tests/base/test_maps.py]: why does this raise?

File: SimPEG/Maps.py (hunk 1 above)
Comment (lheagy) [inline SimPEG/Maps.py]: not necessarily -> you may want to repeat indices.

File: SimPEG/Maps.py (hunk 2 above)
Comment (lheagy) [inline SimPEG/Maps.py]: perhaps that should be implemented in the base maps. We do not currently perform this check on other maps

File: tests/base/test_maps.py (hunk 4 above)
Comment (lheagy) [inline tests/base/test_maps.py]: it is a list

Comment (lheagy) [inline tests/base/test_maps.py]: do we want to support lists?

File: SimPEG/Maps.py (hunk 3 above)
Comment (lheagy) [inline SimPEG/Maps.py]: yep, good call

File: tests/base/test_maps.py (hunk 4 above)
Comment (lheagy) [inline tests/base/test_maps.py]: we should support lists though

**Pull request merged**